Added .mobi (Kindle) Support

emilbryggare opened the pull request:
Support for .mobi format.

* README updated
* Makefile now has ‘make book.mobi’
* Website updated with link to .mobi format.

This works great. It’s simply uses Calibre to convert to ePub to MOBI.

However, before updating http://eloquentjavascript.net/ you need to make sure this file exists on the server http://eloquentjavascript.net/Eloquent_JavaScript.mobi. I could not find anywhere in the code where this file was generate automatically.

marijnh commented on Makefile:
```diff
@@ -118,3 +118,6 @@ epub/hints.xhtml: $(foreach CHAP,$(CHAPTERS),$(CHAP).txt) bin/extract_hints.js
 epubcheck: book.epub
 	epubcheck book.epub 2>&1 | grep -v 'img/.*\.svg'
+
+book.mobi:
```
(This needs book.epub as a dependency after the colon.)

marijnh commented:
Sorry for not taking a look for so long. I'm getting, even with `--remove-first-image`, a bogus page at the start with an apparently random image (one time I got the cat with the arrow from the DOM chapter, one time I got the fractal tree). Do you know what that might be about? If I remove that command line flag, the title page is preserved, but still preceded by a bogus page.

emilbryggare commented:
Thanks for getting back to me.
I've now fixed the dependencies and the cover image.

For the cover image I simply specified the cover image to be used and removed the first image in the book. This worked better. I could not solve it by simple using the .epub book as input. I think this is good enough.

marijnh commented:
Thank you! I've merged this as 1346d1cef4356b3 and set up my build process to build this whenever I push a new commit.
